homeassistant/components/amcrest/binary_sensor.py

```diff
@@ -21,14 +25,19 @@
     BINARY_SENSOR_SCAN_INTERVAL_SECS,
     DATA_AMCREST,
     DEVICES,
-    SENSOR_DEVICE_CLASS,
-    SENSOR_EVENT_CODE,
-    SENSOR_NAME,
     SERVICE_EVENT,
     SERVICE_UPDATE,
 )
 from .helpers import log_update_error, service_signal
 
+
+@dataclass
+class AmcrestSensorEntityDescription(BinarySensorEntityDescription):
+    """Describe Amcrest sensor entity."""
+
+    event_code: str | None = None
```

Comment:
Not sure if this is the best way to associate additional information with the sensor. I could also add a separate key to event code mapping through a separate dictionary.

Reply:
This is good and how it should be solved :)